src/tools/dotnet.lua

```diff
@@ -41,26 +41,51 @@
 		-- Determine the build action for the file, falling back to the file
 		-- extension if no explicit action is available.
 
-		if fcfg.buildaction == "Compile" or ext == ".cs" or ext == ".fs" then
-			info.action = "Compile"
-		elseif fcfg.buildaction == "Embed" or ext == ".resx" then
-			info.action = "EmbeddedResource"
-		elseif fcfg.buildaction == "Copy" or ext == ".asax" or ext == ".aspx" or ext == ".dll" or ext == ".tt" then
-			info.action = "Content"
-		elseif fcfg.buildaction == "Resource" then
-			info.action = "Resource"
-		elseif ext == ".xaml" then
-			if fcfg.buildaction == "Application" or path.getbasename(fname) == "App" then
-				if fcfg.project.kind == p.SHAREDLIB then
-					info.action = "None"
-				else
-					info.action = "ApplicationDefinition"
-				end
-			else
-				info.action = "Page"
-			end
-		else
-			info.action = "None"
+		local actionMap = {
```

Comment:
This and `extensionMap` would probably be better stored outside of the function, and correctly scoped. Being able to override things easily has been one of the best things about the VStudio module.

Reply:
Good point.